### PR TITLE
luci-mod-network: require ssid when join hidden wifi

### PR DESCRIPTION
**Pull request by Ansuel**
WiFi join won't work if is tried to connect to a Hidden Wifi without specifying the WiFi SSID.

Fixes: #2085
Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

### CODE REVIEW

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 1]
```diff
@@ -1911,6 +1911,16 @@ return view.extend({
 				]).then(this.renderContents.bind(this));
 			};
 
+			if (bss.ssid == null) {
+				name = s2.option(form.Value, 'ssid', _('WiFi SSID'), _('SSID must be manually set when a Hidden WiFi is used.'));
```
Comment (jow-) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1915]: Please change to:
`_('Network SSID')` and `_('The correct SSID must be manually specified when joining a hidden wireless network')`

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 2]
```diff
@@ -1911,6 +1911,16 @@ return view.extend({
...
+				name.validate = function(section_id, value) {
```
Comment (jow-) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1916]: I think you can simply replace the entire `name.validate` function with `name.optional = false` or `name.rmempty = false`

**Review by jow- — CHANGES_REQUESTED** [review]

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 3]
```diff
@@ -1911,6 +1911,11 @@ return view.extend({
 				]).then(this.renderContents.bind(this));
 			};
 
+			if (bss.ssid == null) {
+				name = s2.option(form.Value, 'ssid', _('Network SSID'), _('The correct SSID must be manually specified when joining a hidden wireless network'));
+				name.optional = false
```
Comment (jow-) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1916]: Can you add a semicolon here (sorry to be pedandic about that :) )

Comment (Ansuel) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js]: np i'm stupid that i miss think kind of things...

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 4]
```diff
@@ -1911,6 +1911,11 @@ return view.extend({
 				]).then(this.renderContents.bind(this));
 			};
 
+			if (bss.ssid == null) {
```
Comment (jow-) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1914]: I think you also need to actually handle the new `ssid` option somehow around [line 1791](https://github.com/openwrt/luci/pull/4565/files#diff-adf712ab47ca2e5c052d0ad45de5737b2be6860525cd51ec8f424c7e3a43b78aR1791)

**Review by jow- — CHANGES_REQUESTED** [review]

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 5]
```diff
@@ -1790,11 +1790,13 @@ return view.extend({
 		s.handleJoinConfirm = function(radioDev, bss, form, ev) {
 			var nameopt = L.toArray(form.lookupOption('name', '_new_'))[0],
 			    passopt = L.toArray(form.lookupOption('password', '_new_'))[0],
+			    ssidopt = L.toArray(form.lookupOption('ssid', '_new_'))[0],
```
Comment (Ansuel) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1793]: ssid or essid ?

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 6]
```diff
@@ -1837,8 +1839,12 @@ return view.extend({
 					if (bssidval == '1')
 						uci.set('wireless', section_id, 'bssid', bss.bssid);
 				}
-				else if (bss.bssid != null) {
-					uci.set('wireless', section_id, 'bssid', bss.bssid);
+				else {
+					if (ssidval != null)
```
Comment (Ansuel) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1843]: I'm a little confused now... bssid should always available right?
Should I put this if in the bssid check or leave as is? 

Or remove all the check as bssid should be always available and ssidval is mandatory?
Or check both at the same time with an and? (think this should be correct)

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js (hunk 4 above)
Comment (Ansuel) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1920]: Didn't notice that.. Tought the data were saved automagically.

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js (hunk 5 above)
Comment (jow-) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1793]: However you named the option - `'ssid'` in your case.

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js (hunk 6 above)
Comment (jow-) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js:1843]: Leave the BSSID logic as-is, do the SSID stuff in parallel.

**Comment by jow-** [comment]
Merged, thanks @Ansuel !

**Pull request merged**